### ISSUE TITLE
[Gen2 CF-HPG v1.1 Run] Execute preregistered resolution-only seed42 validation experiment

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#47]
## Objective

Execute exactly one preregistered **CF-HPG v1.1 Resolution seed-42 FER2013 train/validation run** after implementation review/merge of PR #46.

Primary hypothesis:

> Increasing only generic raw-patch spatial resolution from v1.0 `6x6 / 64 fine nodes / 16 coarse nodes` to v1.1 `4x4 / 144 fine nodes / 36 coarse nodes`, while freezing graph operator, widths, hierarchy rule, readout, augmentation and optimization, should improve clean-train fit and validation performance if v1.0 underfit was materially caused by coarse spatial compression.

This issue authorizes **one scientific training/validation attempt only**. It does not authorize test access, hyperparameter search, architecture changes, automatic reruns, resume-from-partial, or v1.2.

## Exact authorized source

Run only from exact merged `main` commit:

`934057f60a6cb3bc7ec247e000573a40c390a426`

Reviewed PR #46 head:

`a7b1106d9d8d9802ad38ce26d476cbcc72c29b1c`

Implementation base / accepted v1.0 source:

`562c80fe675d8527416df1b5e441df76f1f62620`

Registered v1.1 model source SHA-256:

`87723409fd6f22d333b9bded2b8707d4e4a3bd5bd3bac19e48022f12f6df0fea`

Registered model inventory:

- exact parameters: `411527`
- trainable variables: `64`
- Keras variables: `74`
- input: `48x48x1`
- non-overlapping patch: `4x4`
- fine grid/nodes: `12x12 = 144`
- raw patch width: `16`
- fine width: `96`, exactly 2 Max-Relative blocks
- hierarchy: exact parameter-free `2x2` arithmetic mean
- coarse grid/nodes: `6x6 = 36`
- coarse width: `128`, exactly 2 Max-Relative blocks
- readout: global mean + global max -> `256 -> 128 -> 7`

No source/model/training-config change is authorized here.

## Locked v1.0 comparison reference

Accepted Issue #44 v1.0 seed42 result:

- clean-train accuracy: `0.5650492876798217`
- clean-train macro-F1: `0.503022788224014`
- validation accuracy: `0.5282808581777654`
- validation macro-F1: `0.46269581824371386`
- accuracy gap: `3.6768429502056277 pp`
- macro-F1 gap: `4.0326969980300165 pp`
- accepted label: `CF_HPG_V1_UNDERFIT`

Exact deltas to report:

- `delta_val_accuracy_pp = 100*(v1.1_val_acc - 0.5282808581777654)`
- `delta_val_macro_pp = 100*(v1.1_val_macro - 0.46269581824371386)`
- `delta_clean_train_accuracy_pp = 100*(v1.1_clean_train_acc - 0.5650492876798217)`
- `delta_clean_train_macro_pp = 100*(v1.1_clean_train_macro - 0.503022788224014)`

## Dataset boundary

Dataset signature:

`fer2013_train28709_val3589_test3589`

Allowed only:

- train: exactly `28,709`
- validation: exactly `3,589`

Absolutely forbidden:

- `test.csv` open/read/hash/load
- test labels/predictions/metrics
- test confusion matrix
- test smoke/debug
- test-based model selection

Do not bypass or rename a forbidden final-split path. Any loader/path guard failure => STOP.

## Exact training configuration

Seed `42`.

Optimizer/training:

- AdamW
- initial LR `3e-4`
- weight decay `5e-4`
- global clipnorm `1.0`
- batch size `64`
- max epochs `100`
- linear warmup `5` epochs
- cosine decay to `1e-6`
- categorical cross-entropy from logits with sparse labels converted internally
- label smoothing `0.05`
- earliest strict maximum `val_accuracy` checkpoint
- early stopping monitor `val_loss`
- patience `15`
- min_delta `0.0`
- no ReduceLROnPlateau

No MixUp, CutMix, SAM, TTA, class weighting, oversampling, alternate optimizer, alternate seed, or outcome-driven config change.

## Exact train-only augmentation

Before patchification, same as v1.0:

1. horizontal flip `p=0.5`
2. x/y translation up to `+/-4 px`
3. rotation `+/-10 deg`
4. contrast `[0.85,1.15]`
5. normalized brightness delta `[-0.10,+0.10]`
6. random erasing `p=0.25`, area `[0.02,0.10]`, aspect `[0.5,2.0]`, zero normalized fill

Geometric interpolation: bilinear.
Fill mode: `REFLECT`.

Validation and clean-train evaluation: augmentation OFF.

## Runtime preflight

Use Kaggle GPU runtime. Before launch record:

- exact repo HEAD
- clean tracked worktree
- model SHA-256
- parameter count
- Python / TensorFlow / Keras versions
- GPU model/count
- CUDA/cuDNN when available
- global mixed-precision policy
- XLA/JIT state if available

Do not modify reviewed source/config to accommodate runtime differences. TensorFlow runtime-internal XLA compilation messages are runtime notes, not authorization to externally enable/monkeypatch XLA.

If the exact command cannot run cleanly, STOP as technical invalid; do not edit model/training protocol.

## Exact authorized command

From exact checkout `934057f60a6cb3bc7ec247e000573a40c390a426`:

```bash
python -m research.candidates.tf_cf_hpg_v1_1_resolution.train_validation_only \
  --train-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --val-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv \
  --output-root /kaggle/working/tf_cf_hpg_v1_1_resolution_seed42
```

Do not add/change flags.

## One-attempt rule

A checkout/auth/dataset/preflight stop **before the training command starts** does not consume the scientific attempt.

Once the training command starts, the one authorized attempt is consumed. After it terminates, crashes, OOMs, is hard-censored, or early-stops:

- STOP
- do not rerun automatically
- do not resume partial checkpoint
- do not change batch/LR/seed/augmentation/architecture
- do not start v1.2

Any second attempt requires explicit research-lead review and a new written amendment justified by technical invalidity, not performance.

## Registered endpoints

Primary:

- validation accuracy at the selected earliest-strict-max-val-accuracy checkpoint

Secondary:

- validation macro-F1
- validation loss
- clean-train accuracy at selected checkpoint
- clean-train macro-F1
- clean-train loss
- accuracy gap pp
- macro-F1 gap pp
- selected epoch
- epochs completed
- exact four deltas vs v1.0
- exact parameter count

Gap definitions:

`acc_gap_pp = 100*(clean_train_accuracy - validation_accuracy)`

`macro_gap_pp = 100*(clean_train_macro_f1 - validation_macro_f1)`

## Frozen decision precedence and thresholds

Precedence is authoritative and was clarified before any v1.1 FER outcome:

1. `CF_HPG_V1_1_STRETCH_PASS`
2. `CF_HPG_V1_1_PASS`
3. `RESOLUTION_OVERFIT_SHIFT`
4. `RESOLUTION_STRONG_SIGNAL`
5. `RESOLUTION_PARTIAL_SIGNAL`
6. `RESOLUTION_UNDERFIT_REMAINS`
7. `RESOLUTION_INCONCLUSIVE`

### `CF_HPG_V1_1_STRETCH_PASS`
All:
- val accuracy `>=0.7000`
- val macro-F1 `>=0.6700`
- both clean gaps `<=8 pp`

### `CF_HPG_V1_1_PASS`
All:
- val accuracy `>=0.6500`
- val macro-F1 `>=0.6200`
- both clean gaps `<=8 pp`

### `RESOLUTION_OVERFIT_SHIFT`
- clean-train accuracy improvement vs v1.0 `>=5.0 pp`
- and either clean gap `>10 pp`

### `RESOLUTION_STRONG_SIGNAL`
- val accuracy improvement vs v1.0 `>=5.0 pp`
- clean-train accuracy improvement vs v1.0 `>=5.0 pp`
- only if no higher-priority label applies

### `RESOLUTION_PARTIAL_SIGNAL`
- val accuracy improvement `>=3.0 pp` and `<5.0 pp`
- only if no higher-priority label applies

### `RESOLUTION_UNDERFIT_REMAINS`
- clean-train accuracy `<0.6500`
- validation accuracy `<0.6000`
- only if no higher-priority label applies

Else `RESOLUTION_INCONCLUSIVE`.

These are single-seed diagnostic/practical labels, not statistical superiority/equivalence claims.

## Required evidence

Preserve the fresh output root without modifying checkpoint contents.

At minimum require:

- `resolved_config.json`
- `model_summary.txt`
- `checkpoints/best_val_accuracy.keras`
- `checkpoints/best_val_accuracy.metadata.json`
- `validation_only_result.json`
- full available training log
- exact command + return code
- runtime/preflight identity
- explicit `test_access=false`

After termination, package the output root into a transport ZIP, compute SHA-256, byte size, member count and integrity. Do not repack after hashing.

If the run is hard-censored before a valid final result exists, preserve partial artifacts as technical evidence only; do not infer the registered scientific label from partial history.

## Interpretation boundary

A PASS/STRETCH would support promoting the architecture, subject to confirmatory seeds and later final-test closure.

A STRONG/PARTIAL signal supports the resolution hypothesis only as a single-seed diagnostic; it does not prove causality/statistical superiority.

`RESOLUTION_UNDERFIT_REMAINS` would strongly motivate moving the next hypothesis from node resolution to the **local visual tokenizer / patch encoder**, not opportunistically sweeping more resolutions.

No test evaluation and no v1.2 before independent evidence review.

## Completion condition

Complete only after:

1. exactly one authorized attempt is executed or technically invalidated;
2. evidence is independently reviewed against this issue;
3. the registered label is accepted or run declared invalid;
4. no test access occurred.

**Comment by Irthn1311** [comment]
INDEPENDENT SCIENTIFIC REVIEW — ACCEPTED.

Reviewed against the uploaded execution artifacts and Issue #47 contract, not the operator summary.

Accepted execution identity:
- exact checkout `934057f60a6cb3bc7ec247e000573a40c390a426`
- clean tracked worktree
- model SHA-256 `87723409fd6f22d333b9bded2b8707d4e4a3bd5bd3bac19e48022f12f6df0fea`
- exact params `411527`
- one authorized attempt, process return code 0
- `scientific_result_valid=true`
- `test_access=false`

Independent archive verification:
- evidence ZIP SHA-256 `4798078eee23d34e36d44f25d2fc977adf55452d0ec2279946149e80bf8dfe8c`
- bytes `4579997`
- members `7`
- outer ZIP integrity PASS
- embedded `.keras` ZIP integrity PASS with 3 internal members

Accepted selected checkpoint:
- zero-based epoch `77` / human epoch 78
- checkpoint val accuracy `0.5734187960624695`
- log inspection confirms this is the earliest observed maximum val-accuracy trajectory point; the run completed all 100 epochs and did not subsequently exceed it.

Accepted endpoint metrics:
- clean-train: accuracy `0.605385070883695`, macro-F1 `0.5474748347135944`, loss `1.1467716292306178`, n=28709
- validation: accuracy `0.5734187796043466`, macro-F1 `0.5154247791468033`, loss `1.2284208524107567`, n=3589
- accuracy gap `3.196629127934836 pp`
- macro-F1 gap `3.205005556679108 pp`

Independently recomputed deltas vs accepted v1.0:
- val accuracy `+4.513792142658124 pp`
- val macro-F1 `+5.272896090308943 pp`
- clean-train accuracy `+4.033578320387332 pp`
- clean-train macro-F1 `+4.445204648958034 pp`

Registered decision is therefore accepted exactly as:
`RESOLUTION_PARTIAL_SIGNAL`

Interpretation boundary: finer patch resolution materially improves both fitting and validation, but the absolute model remains underfit in practical terms (clean-train 60.54%, validation 57.34%). The next hypothesis should target local visual tokenization/patch encoding rather than continuing an opportunistic resolution sweep. No test evaluation and no next scientific run is authorized by this review alone.

Completion condition satisfied; Issue #47 may be closed as completed.
